conformance.adoc

```diff
@@ -528,7 +528,25 @@ The requirements on all other bounds tie point variable attributes are the same
 * An interpolation variable should have 0 dimensions.
 * The recommendations on bounds tie point variable attributes are the same as for bounds variables described in <<cell-boundaries>>.
 
- 
+[[lossy-compression-via-quantization]]
+=== 8.4 Lossy Compression via Quantization
+
+*Requirements:*
+
+* Quantization container variables must have two string-valued attributes, **`algorithm`** and **`implementation`**.
+* The value of **`algorithm`** must be one of the values permitted by this section.
+* Only floating-point type data variables can be quantized. Quantized variables must have and are identified by having a string-valued attribute named **`quantization`**.
+* The value of **`quantization`** must be the name of the quantization container variable which exists in the file.
+* Data variables that were quantized must have an integer type attribute named either **`quantization_nsb`** (for **`algorithm = bitround`**) or **`quantization_nsd`** (for **`bitgroom`**, **`digitround`**, or **`granular_bitround`**).
+* The value of **`quantization_nsb`** must be in the range **`1 \<= NSB \<= 23`** for data type **`float`** or **`real`**, and **`1 \<= NSB \<= 52`** for data type **`double`**.
+* The value of **`quantization_nsd`** must be in the range **`1 \<= NSD \<= 7`** for data type **`float`** or **`real`**, and **`1 \<= NSD \<= 15`** for data type **`double`**.
+
+*Recommendations:*
+
+* The value of **`implementation`** should be specified as _library_ _version_ _number_, e.g., **`libnetcdf version 4.9.2`** or as _client_ _version_ _number_, e.g., **`NCO version 5.2.6`**.
+
+* Data variables that appear in **`formula_terms`** attributes should not be quantized and therefore should not have a **`quantization`** attribute.
+
```

Comment:
Did you mean to remove the recommendation for `implementation` too? The core of that recommendation makes sense to me. Perhaps the `implementation` recommendation should be modified to "..., and then any other information required to disambiguate the source of the algorithm employed."

The resolution of the `formula_terms` recommendation hinges on the outcome of the discussion above regarding `ancillary_variables` of the same intrinsic precision as the variables they refer to.

Reply:
Hi - yes, I did mean this! I was following that because you also state that the `implementation` attribute can also contain any information that is useful, beyond _library_ _version_ _number_, I didn't think it fair to raise a warning if this was done.

Reply:
Ahh, I see. That's fair. My intent was to recommend that the `implementation` attribute **begin** with "_library_ _version_ _number_" or "_client_ _version_ _number_", and then be followed by any additional information. Maybe that's too hard to check?

Reply:
@davidhassell Please respond to this one too. It seems like all conceivable implementation will have either and umambiuous client or library, together with a version. Is a recommendation to put those before any other disambiguating information not worth including? If you think checking that would be too hard or error prone then I agree the recommendation should be removed from the conformance doc.